models/PyCryptoBot.py

```diff
@@ -88,6 +89,24 @@ def to_binance_granularity(granularity: int) -> str:
     return {60: '1m', 300: '5m', 900: '15m', 3600: '1h', 21600: '6h', 86400: '1d'}[granularity]
 
 
+def truncate(f: Union[int, float], n: Union[int, float]) -> str:
```

Comment:
@happz, my linter is freaking out about this line.

Value 'Union' is unsubscriptablepylint(unsubscriptable-object)

Union type; Union[X, Y] means either X or Y.

To define a union, use e.g. Union[int, str]. Details:

The arguments must be types and there must be at least one.

None as an argument is a special case and is replaced by type(None).

Unions of unions are flattened, e.g.::

Union[Union[int, str], float] == Union[int, str, float]

Unions of a single argument vanish, e.g.::

Union[int] == int # The constructor actually returns int

Redundant arguments are skipped, e.g.::

Union[int, str, int] == Union[int, str]

When comparing unions, the argument order is ignored, e.g.::

Union[int, str] == Union[str, int]

You cannot subclass or instantiate a union.

You can use Optional[X] as a shorthand for Union[X, None].


I don't just want to ignore the error in the linter. How can we fix this?

Reply:
I believe this might be a bug in the linter or one of the libraries it's using. See https://github.com/PyCQA/pylint/issues/3882, seems like a false-positive. `Union` is definitely supposed to be used like this, the construction is correct.

How to fix this - there's not much you could do. You can drop the type annotation, or silence the linter. Since the annotation is correct and helpful, personally, I'd silence the linter here, by adding `# noqa` or a similar note, depending on what linter you're using. Plus a link to the upstream issue for the future - once fixed, you could drop the `noqa` bit.

Could you try this, whether it helps on your end?

```python
def truncate(f: Union[int, float], n: Union[int, float]) -> str:  # noqa # https://github.com/PyCQA/pylint/issues/3882
```